tests/ui/question_mark.rs

```diff
@@ -186,6 +186,43 @@ fn f() -> NotOption {
     NotOption::First
 }
 
+#[allow(dead_code)]
+mod issue8288 {
+    struct CustomError;
+
+    fn foo() -> Result<u8, CustomError> {
+        Ok(1)
+    }
+
+    /// Should just replace the if let pattern with `foo()?;`
+    fn bar(some_flag: bool) -> Result<u8, CustomError> {
+        if let Err(err) = foo() {
+            return Err(err);
+        }
+        let mut res = 1_u8;
+        if some_flag {
+            res = 0_u8;
+        }
+        Ok(res)
+    }
+
+    /// No other function logic, same result, replace whole function body with `foo()`
```

Comment:
I'd say that the check for 'same result' for the whole function body is 'too much' - the only case where it will happen is iff the function result can be determined easily and there is no other logic and the two result is exactly equal.

I think we can leave out this specific case.